docs/tutorials/basic/web.md

```diff
@@ -0,0 +1,219 @@
+---
+bodyclass: docs
+layout: docs
+headline: gRPC Basics - Web
+sidenav: doc-side-tutorial-nav.html
+type: markdown
+---
+<p class="lead">This tutorial provides a basic introduction on how to use
+gRPC-Web from browsers.</p>
+
+By walking through this example you'll learn how to:
+
+- Define a service in a .proto file.
+- Generate client code using the protocol buffer compiler.
+- Use the gRPC-Web API to write a simple client for your service.
+
+It assumes a passing familiarity with [protocol
+buffers](https://developers.google.com/protocol-buffers/docs/overview).
+
+<div id="toc"></div>
+
+<a name="why-grpc"></a>
+
+## Why use gRPC?
+
+With gRPC you can define your service once in a .proto file and implement
+clients and servers in any of gRPC's supported languages, which in turn can be
+run in environments ranging from servers inside Google to your own tablet - all
+the complexity of communication between different languages and environments is
+handled for you by gRPC. You also get all the advantages of working with
+protocol buffers, including efficient serialization, a simple IDL, and easy
+interface updating.
+
+
+<a name="setup"></a>
+
+## Define the Service
+
+The first step when creating a gRPC service is to define the service methods
+and their request and response message types using protocol buffers. In this
+example, we define our `EchoService` in a file called
+[`echo.proto`](https://github.com/grpc/grpc-web/blob/master/net/grpc/gateway/examples/echo/echo.proto).
```

Comment:
consider pinning to a particular version instead of master, in case the files move.

Reply:
Fixed.